drivebrain_app/DriveBrainApp.hpp

```diff
@@ -0,0 +1,92 @@
+// DriveBrainApp.hpp
+#pragma once
+
+#include <JsonFileHandler.hpp>
+#include <CANComms.hpp>
+#include <SimpleController.hpp>
+#include <StateEstimator.hpp>
+#include <MCUETHComms.hpp>
+#include <VNComms.hpp>
+#include <MsgLogger.hpp>
+#include <MCAPProtobufLogger.hpp>
+#include <mcap/writer.hpp>
+#include <DrivebrainBase.hpp>
+#include <foxglove_server.hpp>
+#include <DBServiceImpl.hpp>
+
+#include <thread> // std::this_thread::sleep_for
+#include <chrono> // std::chrono::seconds
+#include <condition_variable>
+
+#include <cassert>
+
+
+#include <boost/program_options.hpp>
+#include <boost/asio.hpp>
+
+
+#include <memory>
+#include <optional>
+
+#include <csignal>
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+
+#include <spdlog/spdlog.h>
+#include <atomic>
+
+
+
+class DriveBrainApp {
+public:
+    DriveBrainApp(int argc, char* argv[]);
+    ~DriveBrainApp();
+
+    bool initialize();
+    void run();
+    void stop();
+
+private:
+    
+    std::string getParamPathFromArgs(int argc, char* argv[]);
+    void parseCommandLine(int argc, char* argv[]);
+    void setupSignalHandler();
+    void startThreads();
+    void processLoop();
+    void ioContextLoop();
+    void dbServiceLoop();
```

Comment:
please use all lowercase `snake_case` for all member functions.

also for all private member functions and variables please use `_leading _underscores`

also, please separate the private member functions from the private member variables with two separate `private:` sections in the class

like

```cpp

class Foo {
public:
    Foo() {}
private:
    void _bar();
private:
    int _bar_foo;
};
```